test/c++/frontend_test.cpp

```diff
@@ -91,6 +96,28 @@ TEST_F(FrontendTest, SimpleFunctionWithAdd) {
     EXPECT_EQ (0,  basicTest(is));
 }
 
+TEST_F(FrontendTest, SimpleIntersectionOperator) {
+    istringstream is("element Vertex end\n"
+                     "element Edge end\n"
+                     "const edges : edgeset{Edge}(Vertex,Vertex);\n"
+                     "const vertices1 : vertexset{Vertex} = edges.getVertices();\n"
```

Comment:
Write another unit test in the main function. You can keep this test case. However, in general, the intersection is performed in the main function or user-defined function, and not in the const / global variables.

Reply:
yep!